pkgs/tools/backup/btrbk/default.nix

```diff
@@ -36,11 +53,20 @@ stdenv.mkDerivation rec {
   '';
 
   preFixup = ''
-    wrapProgram $out/sbin/btrbk \
+    wrapProgram $out/bin/btrbk \
       --set PERL5LIB $PERL5LIB \
+      --run 'export program_name=$0' \
```

Comment:
Why was this necessary?

Reply:
`exec -a` does set `$0` for the executed program. If the executed program itself is a (perl or shell) script, it apparently gets overwritten:
```
$ cat p
#!/bin/sh
echo "$0"
$ sh -c "exec -a dollarzero ./p"
/tmp/p
```
lsbtr is a symlink to btrbk, which itself is the wrapper and does `exec -a lsbtr $out/bin/.btrbk-wrapped`, and .btrbk-wrapped is a perl script and sets its $0 to .btrbk-wrapped and cannot find out if it is called as lsblk and behaves as btrbk even when called lsbtr.

My lines work around this and transfer the called executable's name in a separate variable.